lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -0,0 +1,590 @@
+      // <0  a comes before b
+      // >0  b comes before a
+      return aTransaction.transactionNumber - bTransaction.transactionNumber;
+    });
+
+    for (let index = sortedTransactions.length - 1; index >= 0; index--) {
+      const transaction = sortedTransactions[index];
+      const height = transaction.transactionTime;
+      const hash = transaction.transactionTimeHash;
+      if (await this.verifyBlock(height, hash)) {
+        return transaction;
+      }
+    }
+    return;
+  }
+
+  /**
+   * Writes a Sidetree transaction to the underlying Bitcoin's blockchain.
+   * @param anchorFileHash The hash of a Sidetree anchor file
+   */
+  public async writeTransaction (anchorFileHash: string) {
+    console.info(`Anchoring file ${anchorFileHash}`);
+    const sidetreeTransactionString = `${this.sidetreePrefix}${anchorFileHash}`;
+
+    const address = this.privateKey.toAddress();
+    const unspentOutputs = await this.getUnspentCoins(address);
+
+    let totalSatoshis = unspentOutputs.reduce((total: number, coin: Transaction.UnspentOutput) => {
+      return total + coin.satoshis;
+    }, 0);
+
+    const lowBalanceAmount = this.lowBalanceNoticeDays * 24 * 6 * this.bitcoinFee;
+    if (totalSatoshis < lowBalanceAmount) {
+      const daysLeft = Math.floor(totalSatoshis / (24 * 6 * this.bitcoinFee));
+      console.error(`Low balance (${daysLeft} days remaining),\
+ please fund your wallet. Amount: >=${lowBalanceAmount - totalSatoshis} satoshis, Address: ${address.toString()}`);
+    }
+    // cannot make the transaction
+    if (totalSatoshis < this.bitcoinFee) {
+      const error = new Error(`Not enough satoshis to broadcast. Failed to broadcast anchor file ${anchorFileHash}`);
+      console.error(error);
+      throw error;
+    }
+
+    const transaction = new Transaction();
+    transaction.from(unspentOutputs);
+    transaction.addOutput(new Transaction.Output({
+      script: Script.buildDataOut(sidetreeTransactionString),
+      satoshis: 0
+    }));
+    transaction.change(address);
+    transaction.fee(this.bitcoinFee);
+    transaction.sign(this.privateKey);
+
+    if (!await this.broadcastTransaction(transaction)) {
+      const error = new Error(`Could not broadcast transaction ${transaction.toString()}`);
+      console.error(error);
+      throw error;
+    }
+    console.info(`Successfully submitted transaction ${transaction.id}`);
+  }
+
+  /**
+   * Gets all unspent coins of a given address
+   * @param address Bitcoin address to get coins for
+   */
+  private async getUnspentCoins (address: Address): Promise<Transaction.UnspentOutput[]> {
+    const addressToSearch = address.toString();
+    console.info(`Getting unspent coins for ${addressToSearch}`);
+    const requestPath = `/coin/address/${addressToSearch}`;
+
+    const fullPath = new URL(requestPath, this.bitcoinExtensionUri);
+    const response = await this.fetchWithRetry(fullPath.toString());

```

Comment:
Any reason not to use the `rpcCall` method? Same goes to all 3 other occurrences.

Reply:
rpcCall assumes this is a `POST` request, which would fail these particular calls as they are `GET` requests to REST endpoints on the node.